RocksmithToolkitLib/DLCPackage/Manifest2014/Header/AttributesHeader2014.cs

```diff
@@ -129,11 +129,37 @@ public AttributesHeader2014(string arrangementFileName, Arrangement arrangement,
             RouteMask = (int)arrangement.RouteMask;
 
             // TODO: use ManifestFunctions.GetSongDifficulty() method (fix generation algorithm)
-            // TODO: round to 9 decimal places and improve calculation
-            // TODO: fix SongDiff calculations
-            SongDiffEasy = SongContent.SongLength / NotesEasy;
-            SongDiffMed = SongContent.SongLength / NotesMedium;
-            SongDiffHard = SongContent.SongLength / NotesHard;
+            // TODO: improve calculation
+
+            int techCoeff = arrangement.ArrangementPropeties.NonStandardChords +
```

Comment:
Should it be just separate function? I have some space in manifest function class for you :)